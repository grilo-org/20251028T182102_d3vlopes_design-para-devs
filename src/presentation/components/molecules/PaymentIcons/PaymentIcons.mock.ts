import { generateID } from '@/utils/generateID';

import { PaymentIconsProps } from '.';

export const paymentIconsMock: PaymentIconsProps = {
	icons: [
		{
			id: generateID(),
			name: 'Cartão',
			src: 'https://media.graphassets.com/H9mGhU1JSCWlxS7gLttb',
			alt: 'Um quadrado representando um cartão de crédito',
		},
		{
			id: generateID(),
			name: 'Pix',
			src: 'https://media.graphassets.com/FBLLY6c9QgZJWW2Hxl7I',
			alt: 'Um triângulo divido em três partes',
		},
		{
			id: generateID(),
			name: 'Boleto',
			src: 'https://media.graphassets.com/7Jp6jlWdTduM4SKXwrTL',
			alt: 'Um quadrado com pequenos retângulos dentro',
		},
	],
};
