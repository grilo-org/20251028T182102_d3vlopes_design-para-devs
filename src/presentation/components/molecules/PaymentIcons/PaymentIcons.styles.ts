import { styleWrapper } from '@/libs/styles/tailwind-variants';

export const styles = styleWrapper({
	slots: {
		paymentIconPartialWrapperStyles: 'flex items-center gap-2',
		paymentIconPartialImageStyles:
			'relative w-[2.1875rem] h-[2.1875rem] md:w-[2.8125rem] md:h-[2.8125rem]',
		paymentIconPartialTextStyles:
			'font-body font-semibold text-sm inline-block text-transparent bg-clip-text bg-gradient-primary md:text-lg',
	},
});

const makeStyles = styles();

export const classNames = {
	paymentIconPartialWrapperStyles: makeStyles.paymentIconPartialWrapperStyles(),
	paymentIconPartialImageStyles: makeStyles.paymentIconPartialImageStyles(),
	paymentIconPartialTextStyles: makeStyles.paymentIconPartialTextStyles(),
};
