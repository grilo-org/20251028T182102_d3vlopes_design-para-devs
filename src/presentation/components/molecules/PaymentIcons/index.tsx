import { PaymentIconModel } from '@/domain/models/PaymentIcon';

import { PaymentIconPartial } from './partials/PaymentIconPartial';

export interface PaymentIconsProps {
	icons: PaymentIconModel[];
}

export const PaymentIcons = ({ icons }: PaymentIconsProps) => {
	return (
		<div className="flex gap-6 max-w-max">{icons.map(PaymentIconPartial)}</div>
	);
};
