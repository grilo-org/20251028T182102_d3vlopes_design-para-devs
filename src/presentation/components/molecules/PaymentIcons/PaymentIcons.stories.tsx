import type { Meta, StoryObj } from '@storybook/react';

import { paymentIconsMock } from './PaymentIcons.mock';

import { PaymentIcons } from '.';

const meta: Meta<typeof PaymentIcons> = {
	title: 'Molecules/PaymentIcons',
	component: PaymentIcons,
	args: paymentIconsMock,
	parameters: {
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/p1r0o56xHSpG3YHnrNlO3P/LP-Design-para-Devs?type=design&node-id=143-439&mode=design&t=0UEsIP4T0BHN5rrA-0',
		},
	},
};

export default meta;
type Story = StoryObj<typeof PaymentIcons>;

const template: Story = {
	render: (args) => <PaymentIcons {...args} />,
};

export const Default: Story = {
	...template,
};
