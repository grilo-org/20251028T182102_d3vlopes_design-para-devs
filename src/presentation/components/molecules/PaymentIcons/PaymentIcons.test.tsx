import { render, screen } from '@testing-library/react';

import { paymentIconsMock as mock } from './PaymentIcons.mock';

import { PaymentIcons } from '.';

describe('<PaymentIcons />', () => {
	beforeEach(() => render(<PaymentIcons icons={mock.icons} />));

	it('should render correct Image length', () => {
		const imagens = screen.getAllByRole('img');

		expect(imagens).toHaveLength(mock.icons.length);
	});

	it('should render payment method name', () => {
		for (const icon of mock.icons) {
			const paymentMethodName = screen.getByText(icon.name);

			expect(paymentMethodName).toBeInTheDocument();
		}
	});

	it('should render title icons', () => {
		for (const icon of mock.icons) {
			const iconTitle = screen.getByTitle(icon.name);

			expect(iconTitle).toBeInTheDocument();
		}
	});

	it('should match snapshot', () => {
		const { container } = render(<PaymentIcons icons={mock.icons} />);

		expect(container).toMatchSnapshot();
	});
});
