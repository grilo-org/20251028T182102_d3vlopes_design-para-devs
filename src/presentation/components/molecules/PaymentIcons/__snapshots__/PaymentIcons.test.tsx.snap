// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentIcons /> should match snapshot 1`] = `
<div>
  <div
    class="flex gap-6 max-w-max"
  >
    <div
      class="flex items-center gap-2"
    >
      <div
        class="relative w-[2.1875rem] h-[2.1875rem] md:w-[2.8125rem] md:h-[2.8125rem]"
      >
        <img
          alt="Um quadrado representando um cartão de crédito"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FH9mGhU1JSCWlxS7gLttb&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          title="Cartão"
        />
      </div>
      <span
        class="font-body font-semibold text-sm inline-block text-transparent bg-clip-text bg-gradient-primary md:text-lg"
      >
        Cartão
      </span>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="relative w-[2.1875rem] h-[2.1875rem] md:w-[2.8125rem] md:h-[2.8125rem]"
      >
        <img
          alt="Um triângulo divido em três partes"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2FFBLLY6c9QgZJWW2Hxl7I&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          title="Pix"
        />
      </div>
      <span
        class="font-body font-semibold text-sm inline-block text-transparent bg-clip-text bg-gradient-primary md:text-lg"
      >
        Pix
      </span>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="relative w-[2.1875rem] h-[2.1875rem] md:w-[2.8125rem] md:h-[2.8125rem]"
      >
        <img
          alt="Um quadrado com pequenos retângulos dentro"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fmedia.graphassets.com%2F7Jp6jlWdTduM4SKXwrTL&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          title="Boleto"
        />
      </div>
      <span
        class="font-body font-semibold text-sm inline-block text-transparent bg-clip-text bg-gradient-primary md:text-lg"
      >
        Boleto
      </span>
    </div>
  </div>
</div>
`;
